### ISSUE TITLE
gsub does not handle \ consistently

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ThomasDickey/original-mawk#14]
```
See
https://bugs.launchpad.net/ubuntu/+source/mawk/+bug/102208

Also reported by Taylan Ulrich B. as follows:

It appears that inside double-quotes at certain places, backslashes
get cut down.
The behaviour is quite illogical, here's a sample shell session:

% echo | mawk '{ sub(/.*/,"\\"); print }'
\
% echo | mawk '{ sub(/.*/,"\\\\"); print }'
\
% echo | mawk '{ sub(/.*/,"\\\\\\"); print }'
\\
% echo | mawk '{ sub(/.*/,"\\\\\\\\"); print }'
\\
% echo '\\' | mawk '$0 ~ "\\\\\\"'
\\
% echo '\\' | mawk '$0 ~ "\\\\\\\\"'
\\
% echo '\\' | mawk '$0 ~ "\\\\\\\\\\"'

% # (no output, finally)
% echo '\\' | mawk '$0 ~ /\\\\\\/"'


```

Original issue reported on code.google.com by `dic...@his.com` on 28 Jun 2012 at 10:53

**Comment by stephane-chazelas** [comment]
Related: http://austingroupbugs.net/view.php?id=1105

**Comment by ThomasDickey** [comment]
This was fixed in [20200925](https://invisible-island.net/mawk/CHANGES.html#t20200925)
```
        + improve compatibility vs gawk/bwk in gsub handling of backslash by
          only escaping backslash with itself if those backslashes happen to
          precede a "&", e.g., "\\&" vs "\\" (Original-Mawk #14
```
